File: packages/typescript/src/ocrEngine.ts
/**
 * Optional OCR boundary. Bring your own engine — the parser only needs OcrLines.
 *
 * Any object implementing `recognize(imagePath) => Promise<OcrLine[]>` works
 * (Tesseract.js, a cloud OCR API, macOS Vision via tools/vision_ocr.swift…).
 * See examples/ts_tesseract for a reference adapter.
 */

import type { OcrLine } from "./types.js";

export interface OcrEngine {
  recognize(imagePath: string): Promise<OcrLine[]>;
}

File: packages/dart/lib/src/ocr_engine.dart
import 'price_tag_data.dart';

/// Abstraction over the platform OCR engine.
///
/// Production impl wraps `google_mlkit_text_recognition` (on-device).
/// Tests/desktop use fixtures produced by tools/vision_ocr.swift.
/// Either way the output is a list of [OcrLine]s the parser consumes unchanged.
abstract class OcrEngine {
  Future<List<OcrLine>> recognize(String imagePath);
}
